pipeline_dp/dp_engine.py

```diff
@@ -110,3 +114,31 @@ def unnest_cross_partition_bound_sampled_per_key(pid_pk_v):
         return self._ops.flat_map(col,
                                   unnest_cross_partition_bound_sampled_per_key,
                                   "Unnest")
+    
+    def _select_private_partitions(self, col, max_partitions_contributed: int):
+        """Selects and publishes private partitions.
+
+        Args:
+            col: collection, with types for each element: 
+                (partition_key, accumulator: Accumulator)
+            max_partitions_contributed: how many contributed partitions we can get at 
+                most from this operations.
+        
+        Returns:
+            collection of elements (partition_key, accumulator)
+        """
+        budget = self._budget_accountant.request_budget(weight=1, use_eps=True, use_delta=True)
+        
+        def filter_fn(captures: Tuple[Budget, int], row: Tuple[Any, Accumulator]) -> bool:
+            """Lazily create a partition selection strategy and use it to determine which 
```

Comment:
Nit: create -> creates, use -> uses

[Styleguide](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) says "The docstring should be descriptive-style"